pkg/ring/replication_set.go

```diff
@@ -167,8 +167,13 @@ func hasReplicationSetChangedExcluding(before, after ReplicationSet, exclude fun
 		return true
 	}
 
-	sort.Sort(ByAddr(beforeInstances))
-	sort.Sort(ByAddr(afterInstances))
+	if len(beforeInstances) > 0 && len(beforeInstances[0].Tokens) > 0 {
```

Comment:
When does the instance have no token?

Reply:
Just trying to understand if it is possible to have most of instances have token but maybe only one instance doesn't have token. Then checking `len(beforeInstances[0].Tokens) > 0` might still throw index error.

Reply:
I am also ok to modify `ReplicationSet` to include the instance name as well so that we can do a more reliable comparison?

Reply:
You mean as part of `InstanceDesc`?

Reply:
Or the other option would be to have a new function `replicationSet.getTokens()`, sort them and compare. But it's not as efficient as we always have to create long array and sort before comparing.

Reply:
Created new function `ring.GetInstanceDescsForOperation` and `HasInstanceDescsChanged` to solve the problem. It felt like using `ReplicationSet` for the ring change logic was overcomplicating the functions, and all we want to compare is `InstanceDesc` for every `instanceID` in the ring.

The function also does not assume that every instance has tokens, which resolves the original question of this thread.

Updated the PR description accordingly as well.